### DISCUSSION TITLE
Regex - cannot match inline token if it is surrounded by text.

### DISCUSSION THREAD

**Discussion by ivanjaros** [markedjs/marked#3418] [Q&A] [answered]
I have a rating inline token which is `[1-5]*`.
This is my extension:
```js
const rating = {
  name: 'rating',
  level: 'inline',
  start(src) { return src.match(/^[1-5]\*/)?.index; },
  tokenizer(src) {
    const match = src.match(/^([1-5])\*/);
    if (match) {
      return {
        type: 'rating',
        raw: match[0],
        rating: parseInt(match[1]),
      };
    }
  }
};
```

This is the source markdown:
```md
1*

2* suffix

_prefix_ 3*

prefix 4*

prefix 5* suffix
```

This is the result:
![obrázok](https://github.com/user-attachments/assets/770215bc-f331-40d4-a42e-1d0f1256d4f6)

It seems to be unable to work if there is text before it. It works if the text is tokenized(like italic or bold) but not if it is plain text.
I tried to do additional inline parsing but I cannot prepend the text to my own token so this is not working at all.

**Comment by UziTech** [comment]
Your start regex has a ^ so it will only match if the text is at the beginning

**Reply by ivanjaros** [reply]
yes i know. but if i do not use ^ then i will get the content from before the token and i have no way of rendering it. i can render suffix but not prefix.

**Reply by ivanjaros** [reply, marked as answer]
ah, this worked:
```js
const rating = {
  name: 'rating',
  level: 'inline',
  start(src) { return src.match(/[1-5]\*/)?.index; }, <-- no caret
  tokenizer(src) {
    const match = src.match(/^([1-5])\*/); <--- caret
    if (match) {
      return {
        type: 'rating',
        raw: match[0],
        rating: parseInt(match[1]),
      };
    }
  }
};
```
